packages/components/src/panel/types.ts

```diff
@@ -36,3 +41,73 @@ export type PanelRowProps = {
 	 */
 	children: React.ReactNode;
 };
+
+export type PanelBodyProps = {
+	/**
+	 * Props that are passed to the Button component in the PanelBodyTitle within the panel body.
+	 *
+	 * @default {}
+	 */
+	buttonProps?: ButtonProps;
+	/**
+	 * The content to display within the panel body.
+	 *  If the children is a Function, it will be called with an object with the opened property and return its value.
+	 */
+	children?: React.ReactNode | ( ( {} ) => React.ReactNode );
```

Comment:
Thank you for the detailed explanation! Adding a type to an object seemed a bit tricky as the syntax looks like it's the value, so I had a bit of trouble there. But this makes sense!

And I think it's a nice consequence, not to have to import types to the test. 🙂

Reply:
Yeah sometimes the code that we write / suggest can be a bit too "implicit" — in those cases, you can always create some intermediate variables while you're working on it, which may help to break down the problem into smaller bits

<details>
<summary>For example, in this case, the type could be broken down into</summary>

```diff
diff --git a/packages/components/src/panel/types.ts b/packages/components/src/panel/types.ts
index ef2fc65df1..59b44ddde7 100644
--- a/packages/components/src/panel/types.ts
+++ b/packages/components/src/panel/types.ts
@@ -43,6 +43,11 @@ export type PanelRowProps = {
 	children: React.ReactNode;
 };
 
+type ChildrenAsAFunctionProps = { opened: boolean };
+type ChildrenAsAFunction = (
+	props: ChildrenAsAFunctionProps
+) => React.ReactNode;
+
 export type PanelBodyProps = {
 	/**
 	 * Props that are passed to the `Button` component in title within the
@@ -59,9 +64,7 @@ export type PanelBodyProps = {
 	 * The content to display in the `PanelBody`.If a function is provided for
 	 * this prop, it will receive an object with the `opened` prop as an argument.
 	 */
-	children?:
-		| React.ReactNode
-		| ( ( props: { opened: boolean } ) => React.ReactNode );
+	children?: React.ReactNode | ChildrenAsAFunction;
 
 	/**
 	 * The CSS class to apply to the wrapper element.
```

</details>